virttest/qemu_vm.py

```diff
@@ -548,17 +548,26 @@ def process_sandbox(devices, action):
                 return sandbox_option
             if action == "add":
                 sandbox_option = " -sandbox on"
+                _params = {
+                    "elevateprivileges": params.get(
+                        "qemu_sandbox_elevateprivileges", "deny"
+                    ),
+                    "obsolete": params.get("qemu_sandbox_obsolete", "deny"),
+                    "resourcecontrol": params.get(
+                        "qemu_sandbox_resourcecontrol", "deny"
+                    ),
+                    "spawn": params.get("qemu_sandbox_spawn", "deny"),
+                }
             elif action == "rem":
                 sandbox_option = " -sandbox off"
-            _params = {
-                "elevateprivileges": params.get("qemu_sandbox_elevateprivileges"),
-                "obsolete": params.get("qemu_sandbox_obsolete"),
-                "resourcecontrol": params.get("qemu_sandbox_resourcecontrol"),
-                "spawn": params.get("qemu_sandbox_spawn"),
-            }
+                _params = {
+                    "elevateprivileges": params.get("qemu_sandbox_elevateprivileges"),
+                    "obsolete": params.get("qemu_sandbox_obsolete"),
+                    "resourcecontrol": params.get("qemu_sandbox_resourcecontrol"),
+                    "spawn": params.get("qemu_sandbox_spawn"),
+                }
```

Comment:
This is what I was thinking about, although we have some dup code, we cover both situations, i.e. when sandbox is disabled, there is also a way for VT to set the options in case we'd like to enhance it some day in future.

But refer to the qemu code, all options are ignored if sandbox is disabled. So it's safe to keep the original change too.

In summary, both code changes are OK for me